src/charm.py

```diff
@@ -534,24 +571,8 @@ def _compare_secret_ids(secret_id1: str, secret_id2: str) -> bool:
             return pure_id1 == pure_id2
         return False
 
-    # BEGIN: charm events
-    def _on_mongod_pebble_ready(self, event) -> None:
-        """Configure MongoDB pebble layer specification."""
-        # Get a reference the container attribute
-        container = self.unit.get_container(Config.CONTAINER_NAME)
-        if not container.can_connect():
-            logger.debug("mongod container is not ready yet.")
-            event.defer()
-            return
-
-        # We need to check that the storages are attached before starting the services.
-        # pebble-ready is not guaranteed to run after storage-attached so this check allows
-        # to ensure that the storages are attached before the pebble-ready hook is run.
-        if any(not storage for storage in self.model.storages.values()):
-            logger.debug("Storages are not attached yet")
-            event.defer()
-            return
-
+    def __filesystem_handler(self, container: Container) -> bool:
```

Comment:
mind adopting raise format, rather than the return a bool format?

Reply:
Sure